extensions/binding/org.eclipse.smarthome.binding.sonos/src/main/java/org/eclipse/smarthome/binding/sonos/internal/handler/ZonePlayerHandler.java

```diff
@@ -2889,7 +2878,10 @@ public String getCurrentZoneName() {
     @Override
     public void onStatusChanged(boolean status) {
         if (status) {
-            updateStatus(ThingStatus.ONLINE);
+            if (getThing().getStatus() != ThingStatus.ONLINE) {
+                updateStatus(ThingStatus.ONLINE);
```

Comment:
I think you should let the polling job set the thing ONLINE through the call to updatePlayerState. That way the thing would be set to ONLINE in one unique place.
By the way, that is a good idea to run the job immediately after the device is found through UPnP, it can make the thing going ONLINE a little faster.

Reply:
At the same time, that means you could generate a conflict between this one time job and the regular scheduled job, in particular each one requesting GENA subscriptions.

Reply:
As there is a lock implemented for GENA subscriptions, it should be finally probably ok.